packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -130,16 +157,17 @@ export class NodeSDK {
 
   /**
    * Once the SDK has been configured, call this method to construct SDK components and register them with the OpenTelemetry API.
+   * SDK can be used as
+   * @example
+   *
+   *     sdk.start();
+   *     // sdk ready
    */
-  public async start() {
-    if (this._autoDetectResources) {
-      await this.detectResources();
-    }
-
+  public start(): void {
     if (this._tracerProviderConfig) {
       const tracerProvider = new NodeTracerProvider({
         ...this._tracerProviderConfig.tracerConfig,
-        resource: this._resource,
+        resource: (this._detectResources() as unknown) as Resource,
```

Comment:
This type should allow `Promise<Resource>`. Only `ReadableSpan` type should have to disallow `Promise`. Then you won't have to cast here.